foodsaving/groups/api.py

```diff
@@ -98,9 +99,10 @@ def partial_update(self, request, *args, **kwargs):
         return super().partial_update(request, *args, **kwargs)
 
     def get_queryset(self):
+        # can join all groups but only do anything else to groups where user is member of
         if self.action == 'join':
             return self.queryset
-        return self.queryset.filter(members=self.request.user)
+        return self.queryset.filter(groupmembership__user=self.request.user, groupmembership__roles__contains=[APPROVED])
```

Comment:
I guess this is one of the bits that needs to be fiddled with in future. Non-approved users should have some access to the groups they applied to, but not full access.

Reply:
yep. But as the data in the serializer may also change, I see no reason for doing something there yet.